packages/core/src/types.ts

```diff
@@ -1794,11 +1789,18 @@ export type ActorRefFrom<T> = ReturnTypeOrValue<T> extends infer R
       >
     : R extends Promise<infer U>
     ? ActorRef<{ type: string }, U | undefined>
-    : R extends Behavior<infer TEvent, infer TSnapshot>
-    ? ActorRef<TEvent, TSnapshot>
+    : R extends AnyBehavior
```

Comment:
The previous version was slightly better because even with the patch from [this PR](https://github.com/statelyai/xstate/pull/3579) we won't be able to match `Behavior<never, ...>` here:
[TS playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgKIDcLgPICMBWECYyA3gFDLJgCeADhAFzIDOYUoA5gNzkC+5cqEixEKAIIgaGLGDyFiyCAA9IIACYs0mHASIkKVANoBrCDWZsOITgF1mcKbwHlaDZJJoAhCAAs46MAA9lDIALzIPv6BIQA8jjQANMgJAHy8QuDQ8EiRfgHBULEAKjLgSqpYmmTU9Eys7FzIfMnFAMogcHQsvkFgqWSU1FCOLMBgwSDMABQAthAsLHCc9aU6YACU4QPtnd29YM6CwtliHsQhAEoQMCVlJCpq1aS1DJaNNs2tHV09fQOGVhVGbzRbLVb3LZhAboILAdS8KgrMA-fZ9GZQnaov6HfiCNwSC5Qa4wABiUCCsyiBTixQGEWKFSeWmpMSKoBg0GQa1kyQ5XN2vwOqSGAH5zmArjc7utvnscSKqMwQBBMFBeASJVKyRTZiV6dymVUtJ5WYVkOLxESSeTKWbaQNlaroBl1EQADZwKAoBBBEBsZC4ACMzHtRRVauSL19AFdwMwg810q46h5ExErZLiTdbXqCUEYIGg8mAPQlqjIAB6osEvv9JD+AHdUFAKVBmKb8mzwkWgA)

I'm not quite sure how useful it is to actually ever end up with `never` here, but the previous version was handling this case without breaking anything.

Reply:
f88053111e76692b11f99d512c177a5f6dd5b0bb